### PR TITLE
[MPLUGIN-395] Upgrade Maven Reporting API/Impl to 3.1.0

### PR DESCRIPTION
**Pull request by michael-o**
Implicitly update MSITE to 3.11.0 and MPIR 3.2.2 for consistency with reporting.

This closes #74

### CODE REVIEW

File: maven-plugin-plugin/pom.xml [hunk 1]
```diff
@@ -350,7 +350,8 @@
               <settingsFile>src/it/settings.xml</settingsFile>
               <streamLogsOnFailures>true</streamLogsOnFailures>
               <filterProperties>
-                <sitePluginVersion>3.3</sitePluginVersion>
+                <sitePluginVersion>3.11.0</sitePluginVersion>
+                <projectInfoReportsPlugin>3.2.2</projectInfoReportsPlugin>
```
Comment (slawekjaranowski) [inline maven-plugin-plugin/pom.xml:354]: Why not use the same version like in project?
Do we want later use another version in tests then in project?
I afraid that next we update version in project but not for tests.
And how to managed version if we have plugin version from parent?

Comment (michael-o) [reply maven-plugin-plugin/pom.xml:354]: In most cases reactor plugins and IT plugins are decoupled. Look at other projects. Moreover, I did not want to rewrite the entire thing. You have valid concerns, but they need to be addressed on a global level for all components.

Comment (slawekjaranowski) [reply maven-plugin-plugin/pom.xml:354]: ok, agreed

**Comment by slachiewicz** [comment]
Maybe we should have separate upgrades of  one plugin and independent upgrade of it tests? To easier detect if plugin updates not affects old it tests?

**Comment by michael-o** [comment]
> Maybe we should have separate upgrades of one plugin and independent upgrade of it tests? To easier detect if plugin updates not affects old it tests?

I totally agree with you and will explain why I did this in one go:
* Both API/Impl go hand on hand for Doxia 2 preparation
* When you update both reporting components ITs start to fail

Basically spilitting up is posslbe, but will leave commits in broken state that is why I have them together. In other projects I have upgraded ITs completely separately.

**Review by slachiewicz — APPROVED** [review]

**Pull request closed without merging**